apps/designer/app/canvas/shared/use-drag-drop.ts

```diff
@@ -53,6 +50,16 @@ export type DragEndPayload = {
 
 export type DragMovePayload = { canvasCoordinates: Point };
 
+const getInstanceElementById = (id: Instance["id"]) => {
+  return document.querySelector(`[${idAttribute}="${id}"]`);
+};
+
+const getInstanceIdFromElement = (
+  element: Element
+): Instance["id"] | undefined => {
+  return element.getAttribute(idAttribute) ?? undefined;
+};
```

Comment:
I'd keep `findInstanceByElement, getInstanceElementById, getInstanceIdFromElement` in utils.

Would only change `findInstanceByElement`'s argument from `rootInstance` to `instancesById`

Reply:
Do we need them In utils if never reuse?

Reply:
Oleg asked to put in utils originally. But I kinda agree that it makes sense to have these little self contained potentially re-usable pieces of code in utils